src/axom/primal/operators/detail/winding_number_2d_impl.hpp

```diff
@@ -284,6 +286,75 @@ double convex_endpoint_winding_number(const Point<T, 2>& q,
+ * \param [in] isOnEdge An returned flag if the point is on the curve
+ * \param [in] edge_tol The physical distance level at which objects are considered indistinguishable
+ * \param [in] EPS Miscellaneous numerical tolerance level for nonphysical distances
+ *
+ * Computes the GWN by decomposing into rational Bezier curves
+ *  and summing the resulting GWNs. Far-away curves can be evaluated
+ *  without decomposition using direct formula.
+ * 
+ * \return The GWN.
+ */
+template <typename T>
+double bezier_winding_number_memoized(const Point<T, 2>& q,
+                                      const NURBSCurveGWNCache<T>& nurbs_cache,
+                                      int bezier_idx,
+                                      int refinement_level,
+                                      int refinement_index,
+                                      bool& isOnCurve,
+                                      double edge_tol = 1e-8,
+                                      double EPS = 1e-8)
+{
+  // Early exit for degenerate curves
+  const int deg = nurbs_cache.getDegree();
+  if(deg <= 0)
+  {
+    return 0.0;
+  }
+
+  auto& bezier_data = nurbs_cache.getSubdivisionData(bezier_idx, refinement_level, refinement_index);
+  auto& bezier_curve = bezier_data.curve;
+
+  // If outside a bounding box, the curve can be treated as linear between its endpoints
+  if(!bezier_data.boundingBox.contains(q) || bezier_curve.isLinear(EPS))
+  {
+    return detail::linear_winding_number(q, bezier_curve[0], bezier_curve[deg], isOnCurve, edge_tol);
+  }
+
+  // If the control polygon is convex, we can handle coincidence with a direct formula.
+  //  In this case, accessing subdivision bounding boxes is fast enough to skip checking a polygon
+  if(bezier_data.isConvexControlPolygon &&
+     (squared_distance(q, bezier_curve[0]) <= edge_tol * edge_tol ||
+      squared_distance(q, bezier_curve[deg]) <= edge_tol * edge_tol))
+  {
+    isOnCurve = true;
+    return convex_endpoint_winding_number(q, bezier_curve, edge_tol, EPS);
+  }
+
+  return detail::bezier_winding_number_memoized(q,
+                                                nurbs_cache,
+                                                bezier_idx,
+                                                refinement_level + 1,
+                                                2 * refinement_index,
+                                                isOnCurve,
+                                                edge_tol,
+                                                EPS) +
+    detail::bezier_winding_number_memoized(q,
+                                           nurbs_cache,
+                                           bezier_idx,
+                                           refinement_level + 1,
+                                           2 * refinement_index + 1,
+                                           isOnCurve,
+                                           edge_tol,
+                                           EPS);
```

Comment:
Aside: Weird formatting on the `+`

Reply:
Yes, please assign to temporaries and return the sum.